core/src/main/java/org/arquillian/smart/testing/configuration/Configuration.java

```diff
@@ -0,0 +1,279 @@
+import org.arquillian.smart.testing.hub.storage.local.LocalStorageFileAction;
+import org.yaml.snakeyaml.Yaml;
+
+import static org.arquillian.smart.testing.report.SmartTestingReportGenerator.REPORT_FILE_NAME;
+import static org.arquillian.smart.testing.scm.ScmRunnerProperties.DEFAULT_LAST_COMMITS;
+import static org.arquillian.smart.testing.scm.ScmRunnerProperties.HEAD;
+import static org.arquillian.smart.testing.scm.ScmRunnerProperties.SCM_LAST_CHANGES;
+import static org.arquillian.smart.testing.scm.ScmRunnerProperties.SCM_RANGE_HEAD;
+import static org.arquillian.smart.testing.scm.ScmRunnerProperties.SCM_RANGE_TAIL;
+
+public class Configuration {
+
+    private static final Logger logger = Logger.getLogger();
+
+    public static final String DEFAULT_MODE = "selecting";
+    public static final String DEFAULT_STRATEGIES = "";
+    public static final String SMART_TESTING_REPORT_ENABLE = "smart.testing.report.enable";
+
+    public static final String SMART_TESTING = "smart.testing";
+    public static final String SMART_TESTING_MODE = "smart.testing.mode";
+    public static final String SMART_TESTING_APPLY_TO = "smart.testing.apply.to";
+    public static final String SMART_TESTING_VERSION = "smart.testing.version";
+    public static final String SMART_TESTING_DISABLE = "smart.testing.disable";
+    public static final String SMART_TESTING_DEBUG = "smart.testing.debug";
+
+    public static final String SMART_TESTING_YML = "smart-testing.yml";
+    public static final String SMART_TESTING_YAML = "smart-testing.yaml";
+
+    private String[] strategies = new String[0];
+    private RunMode mode;
+    private String applyTo;
+
+    private boolean disable;
+    private boolean debug;
+
+    private Report report;
+    private Scm scm;
+
+    public String[] getStrategies() {
+        return strategies;
+    }
+
+    public void setStrategies(String... strategies) {
+        this.strategies = strategies;
+    }
+
+    public RunMode getMode() {
+        return mode;
+    }
+
+    public void setMode(RunMode mode) {
+        this.mode = mode;
+    }
+
+    public String getApplyTo() {
+        return applyTo;
+    }
+
+    public void setApplyTo(String applyTo) {
+        this.applyTo = applyTo;
+    }
+
+    public boolean isDisable() {
+        return disable;
+    }
+
+    public void setDisable(boolean disable) {
+        this.disable = disable;
+    }
+
+    public boolean isDebug() {
+        return debug;
+    }
+
+    public void setDebug(boolean debug) {
+        this.debug = debug;
+    }
+
+    public Report getReport() {
+        return report;
+    }
+
+    public void setReport(Report report) {
+        this.report = report;
+    }
+
+    public Scm getScm() {
+        return scm;
+    }
+
+    public void setScm(Scm scm) {
+        this.scm = scm;
+    }
+
+    public static Configuration load() {
+        final File parent = Paths.get("").toAbsolutePath().toFile();
```

Comment:
I would be great to have option to define the root dir as a parameter of the method `load()`